addons/account/models/account_tax.py

```diff
@@ -268,6 +268,19 @@ def _check_children_scope(self):
             ):
                 raise ValidationError(_('Nested group of taxes are not allowed.'))
 
+    @api.constrains('amount_type')
+    def _check_parent_type(self):
+        for tax in self:
+            if tax.amount_type == 'group':
+                group_parent_taxes = self.env['account.tax'].search([
+                    ('amount_type', '=', 'group')
+                ]).filtered(lambda parent_tax: tax in parent_tax.children_tax_ids)
```

Comment:
I'd really prefer to avoid the `search` within a loop.
I'm wondering if it wouldn't be a better idea to add the "inverse" field of the `children_tax_ids` to get the parents from the tax. And let the ORM handle the thing.
I'll do that when I come back from holidays early in January and keep you updated. :slightly_smiling_face: